markdown_lab_rs.py

```
@@ -41,19 +53,175 @@ def convert_html_to_markdown(html: str, base_url: str = "") -> str:
+    document = parse_markdown_to_document(markdown_content, base_url)
+
+    if output_format == OutputFormat.JSON:
+        return json.dumps(document, indent=2)
+    if output_format == OutputFormat.XML:
+        return document_to_xml(document)
+
+    # Fallback to markdown if format not recognized
+    return markdown_content
+
+
+def parse_markdown_to_document(markdown: str, base_url: str) -> Dict:
```

Ruff (0.8.2)

`parse_markdown_to_document` is too complex (13 > 10)

(C901)